### DISCUSSION TITLE
Little bit confused, is there an option to host your own server?

### DISCUSSION THREAD

**Discussion by Star-X555** [open-duelyst/duelyst#252] [Q&A] [answered]
Asking as I'd honestly prefer to muck about offline before committing to any kind of online TCG progression.  Also I'm spoiled by Tabletop Sim having literally every Magic the Gathering/Pokemon/WoW TCG/whatever card available simply by googling the card(s) you want and plopping the pictures into TTS, so being in full control of what's actually available is a big deal.

I did read that Staging is by default set up to function like TTS "deckbuilder" workshop entries in that everything's unlocked immediately, no grinding or progression required, my question is more related to "can I host the damn server locally on something like my Steam Deck so I can play over LAN on the go with friends and/or family" rather than whether the "official servers" offer unlocking everything immediately.

On a side note, is hotseat multiplayer possible in the current build, and if not, would it be feasible to implement it on an unofficial server if one will be available?  Given the existence of web-based interfaces it might be moot as virtually any device can run those, but it'd still be nice to be able to just pass my Steam Deck around between family members for playing locally without requiring running multiple instances AND the server application.

**Comment by willroberts** [comment, marked as answer]
> Is there an option to host your own server?

Yes, check out the documentation: https://github.com/open-duelyst/duelyst/blob/main/docs/infrastructure/README.md
  
> can play over LAN  
  
There is no LAN functionality, but you could take a look at what would be required to add it. There is currently a dependency on Firebase which would need to be replaced.  
  
> On a side note, is hotseat multiplayer possible in the current build, and if not, would it be feasible to implement it on an unofficial server if one will be available  
  
This is not implemented. I'm not sure how much effort would be required to add it.

**Reply by Star-X555** [reply]
> > Is there an option to host your own server?
> 
> Yes, check out the documentation: https://github.com/open-duelyst/duelyst/blob/main/docs/infrastructure/README.md
> 

Cool, may muck with it later :)

> > can play over LAN
> 
> There is no LAN functionality, but you could take a look at what would be required to add it. There is currently a dependency on Firebase which would need to be replaced.
> 

To be fair, if you're hosting the server on a LAN, technically any devices that are a part of that LAN can connect to that server without direct internet access.  Well, outside of a "phone home" requirement that only works on the internet.  In any event a server running on a suitably powerful machine on a LAN should allow any capable devices to connect to it and play together.


> > On a side note, is hotseat multiplayer possible in the current build, and if not, would it be feasible to implement it on an unofficial server if one will be available
> 
> This is not implemented. I'm not sure how much effort would be required to add it.

Yeah, figured as such.  Still not a huge problem as I can likely run the server on one of my portable Linux devices and have my other handhelds connect to the LAN IP address with their browsers to "bypass" this issue.  That or just multibox with several separate Firefox instances on one device if it's feasible.

Will be playing around with this at some point and experimenting with having my tablet act as an access point for my family's devices to play together outside of wifi access, like when on a trip :)
